### DISCUSSION TITLE
Deploying releases via tags / CI

### DISCUSSION THREAD

**Discussion by vemv** [clojure-emacs/cider#3130] [Ideas]
I'm wondering if we should start deploying releases by simply `git push --tags`, and let a well-configured CI do the deployment.

The pros would be:

* fewer things that can go wrong
  * releases can be borked depending on one's laptop (e.g. wrong JDK, dirty env, dirty git status, wrong branch, etc)
* less trust needed
  * pushing a tag needs less trust than granting Clojars creds
* we can afford to have a slower deploy step
  * if the deploy is performed by CI, one doesn't have to be staring at it
  * so we can e.g. disable mranderson parallelism, and/or not care about any associated slowness

This is how it's done in Eastwood which has worked well for us for a while:

https://github.com/jonase/eastwood/blob/afae433986873f9f22ec39a60409ab1a597348ae/.circleci/config.yml#L271-L325
https://github.com/jonase/eastwood/blob/afae433986873f9f22ec39a60409ab1a597348ae/.circleci/maybe_deploy.clj

It could be further made idiomatic for our practices, e.g. `lein` instead of `clojure`, and also use `make` somewhere in the process :)

WDYT? @bbatsov

**Comment by bbatsov** [comment]
I guess that's for the Clojure projects (e.g. `cider-nrepl`), not CIDER itself, right? Is so - sure, I think that's a good idea. I was just too lazy/busy to automate the release/deployment process.

**Reply by vemv** [reply]
> I guess that's for the Clojure projects (e.g. cider-nrepl), not CIDER itself, right?

Yep

**Reply by vemv** [reply]
> , I think that's a good idea.

Alright, let's go for it, will start with refactor-nrepl since it's a slow-moving target

**Comment by vemv** [comment]
As an additional note, here's an overview of what's implemented in eastwood https://archive.is/BB9Gk and how to set up signed releases https://archive.is/oNutE

I believe signed releases make no sense here, because if member can push a git tag, then what does a _shared_ signature mean?
